source/geod24/LocalRest.d

```diff
@@ -148,13 +148,227 @@ private enum Status
+        else
+            ret = this.command_chn.write(Variant(msg));
+
+        return ret;
+    }
+
+    /*******************************************************************************
+
+        Get a unique id for a `Command` to be sent from this `Connection`
+
+        Returns:
+            Unique Command id
+
+    *******************************************************************************/
+
+    size_t getNextCommandId () @safe
+    {
+        return this.next_cmd_id++;
+    }
+
+    /*******************************************************************************
+
+        Wait for a `Response` with specific id
+
+        Params:
+            resp_id = Response id to wait for
+            timeout = optional timeout duration for the operation
+
+        Returns:
+            Response
+
+    *******************************************************************************/
+
+    Response waitResponse (size_t resp_id, Duration timeout = Duration.init) @trusted
+    {
+        if (isMainThread())
+        {
+            Response res;
+            scheduler.start({
+                // Loop until we get the Response we are looking for
+                while (this.resp_chn.read(res, timeout))
+                    if (res.id == resp_id)
+                        return;
+                res = Response(resp_id, Status.Timeout, SerializedData("Request timed-out"));
+            });
+            return res;
+        }
+        else
+        {
+            // Response may already be ready
+            auto existing_res = (resp_id in this.waiting_list);
+            if (existing_res !is null)
+                return existing_res.res;
+
+            // Block while waiting the Response
+            auto blocker = C.thisScheduler().new FiberBlocker();
+            this.waiting_list[resp_id] = Waiting(blocker);
+            if (blocker.wait(timeout))
+                return this.waiting_list[resp_id].res;
+            else
+                return Response(resp_id, Status.Timeout, SerializedData("Request timed-out"));
+        }
+    }
+
+    /*******************************************************************************
+
+        Notify the task waiting for a Response
+
+        Params:
+            res = Newly arrived Response
+
+    *******************************************************************************/
+
+    void notifyWaiter (Response res)
+    {
+        // Is the waiting Fiber already blocked?
+        auto waiter = (res.id in this.waiting_list);
+        if (waiter !is null)
+        {
+            waiter.res = res;
+            waiter.blocker.notify();
+        }
+        else // Fiber is not yet blocked, create an entry for the related Fiber to use
+            this.waiting_list[res.id] = Waiting(null, res);
+    }
+
+    /*******************************************************************************
+
+        Close the `Channel`s associated with this `Connection`
```

Comment:
It would be good to document what happens after the channel is closed. Do the other nodes's requests fail, or time out?